File: Python/personaje.py
class Personaje:
  def __init__(self, nombre, vida, ataque):
    self.nombre = nombre
    self.vida = vida
    self.ataque = ataque
    def atacar(self, otro_personaje):
     otro_personaje.vida = otro_personaje.vida - self.ataque
     print(f"{self.nombre} ataca a {otro_personaje.nombre}")
     print(f"Vida de {otro_personaje.nombre}:{otro_personaje.vida}")

# Ejemplo de uso
heroe = Personaje("Link" , 100, 15)
villano = Personaje("Ganon" ,120, 10)

heroe.atacar(villano)

File: Java/Personaje.java
public class Personaje {
  private String nombre;
  private int vida;
  private int armadura;

  public Personaje(String nombre, int vida, int armadura) {
    this.nombre = nombre;
    this.vida = vida;
    this.armadura = armadura;

  }

  public void atacar(Personaje enemigo) {
    System.out.printIn(this.nombre + "atacó a " enemigo.nombre):
    enemigo.vida = enemigo.vida - 10:
      
  }

  public void defender() {
    this.armadura = this.armadura + 8:
    System.out.printIn(this.nombre + "se defendió. armadura: " + this.armadura);

  }

  public boolean esta_vivo() {
    return this.vida > 0:

  }
